### DISCUSSION TITLE
Use 4D nifti volumes to train a 3D network in multichannel

### DISCUSSION THREAD

**Discussion by anais2390** [Project-MONAI/MONAI#3089] [Q&A] [answered]
Hi !
I am currently working on lesion segmentation and I would like to train a 3D multichannel model using 2 differents MRI modality volumes.
I a previous work, the volumes of the 2 differents modality have been coregistred, concatenated and saved into a unique 4D volumes in nifti format. 

My question is : is there a way to load the 4D volumes using LoadImage and to use a Transform to convert the 4D volume into a 3D volume with 2 channels ? 

I would have the same question for 2D multichannel segmentation using 3D nifti volumes ? 

Thank you for your help, please let me know if you need more details or if the question is not clear enough.

**Comment by Nic-Ma** [comment]
Hi @anais2390 ,

Thanks for your interest here.
Could you please help provide an example image shape and the target image shape?

Thanks.

**Reply by anais2390** [reply]
Hi @Nic-Ma  thanks for your reply.

**In the 3D multichannel case**  : we have nifti 4D volumes with shape _(nb_pixel_x, nb_pixel_y, nbslices, nbchannels)_. Typicall example  is shape _(256, 256, 100, 4)_.
**In the 2D multichannel case** : we have nifti 3D volumes with shape _(nb_pixel_X, nb_pixel_y, nbchannels)._ Typically _(256, 256, 4)_.

Thanks for your help!

**Comment by Nic-Ma** [comment]
Hi @anais2390 ,

MONAI can load the 4D data to a 4D array, I don't quite understand your request "convert the 4D volume into a 3D volume with 2 channels", could you please help share more details?

Thanks.

**Reply by anais2390** [reply]
I am sorry, that was not clear, I will try to give you an example : 

We are working on 3D segmentation in MRI volume data. But we would like to perform this segmentation using several sequences (T1, T2 for exemple). So during the database elaboration we concatenate the 3D T1 and the 3D T2 into one nifti 4D file. 
Then, we load the data in the DataLoader: the 4D nifti is loaded but it is recognized as a 4D data with 1 channel. We would like to be recognize as a 3D data with 2 channel (ie, the 4th dimension is in reality the channel dimension). 

Is it more clear ? 

In the 2D case it is the same problem : We concatenate the 2D slices of T1 and T2 into one 3D nifti file and we would like to be read as a 2D data with the 3rd dimension being the channel dimension.

**Comment by dongyang0122** [comment, marked as answer]
Hi @anais2390 , "LoadImageD" supports loading 4D images or stacks of 3D images. After loading, you need make sure the input arrays are channel-first for other operations. The following example is the brain tumor segmentation with multi-modal input, which is very similar to your application.

https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/brats_segmentation_3d.ipynb

**Reply by anais2390** [reply]
Thanks @dongyang0122 , the feature I was looking for was the `EnsureChannelFirst()` transform
